ape_safe/client.py

```diff
@@ -165,72 +271,158 @@ def _all_transactions(self) -> Iterator[dict]:
-        """
-        confirmed: Confirmed if True, not confirmed if False, both if None
-        """
-        next_nonce = self.get_next_nonce()
+    def get_confirmations(self, safe_tx_hash: SafeTxID) -> Iterator[SafeTxConfirmation]:
+        url = (
+            f"{self.transaction_service_url}/api"
+            f"/v1/multisig-transactions/{str(safe_tx_hash)}/confirmations"
+        )
+        while url:
+            response = requests.get(url)
+            if not response.ok:
+                raise ClientResponseError(url, response)
 
-        for txn in self._all_transactions():
-            if txn["nonce"] < starting_nonce:
-                break  # NOTE: order is largest nonce to smallest, so safe to break here
+            data = response.json()
+            yield from map(SafeTxConfirmation.parse_obj, data["results"])
+            url = data["next"]
 
-            isConfirmed = len(txn["confirmations"]) >= txn["confirmationsRequired"]
-            isExecuted = "isExecuted" in txn and txn["isExecuted"]
+    def post_transaction(self, safe_tx: SafeTx, sigs: Dict[AddressType, MessageSignature]):
+        tx_data = UnexecutedTxData.from_safe_tx(safe_tx, self.safe_details.threshold)
+        tx_data.signatures = HexBytes(
+            reduce(
+                lambda raw_sig, next_sig: raw_sig + next_sig.encode_rsv(),
+                order_by_signer(sigs),
+                b"",
+            )
+        )
+        post_dict: Dict = {}
+        for key, value in tx_data.dict().items():
+            if isinstance(value, HexBytes):
+                post_dict[key] = value.hex()
+            elif isinstance(value, OperationType):
+                post_dict[key] = int(value)
+            elif isinstance(value, datetime):
+                # not needed
+                continue
+            else:
+                post_dict[key] = value
 
-            if confirmed is not None:
-                if not confirmed and isExecuted:
-                    break  # NOTE: Break at the first executed transaction
+        url = f"{self.transaction_service_url}/api/v1/safes/{tx_data.safe}/multisig-transactions"
+        json_data = {"origin": "ApeWorX/ape-safe", **post_dict}
+        response = requests.post(url, json=json_data)
 
-                elif confirmed and not isConfirmed:
-                    continue  # NOTE: Skip not confirmed transactions
+        if not response.ok:
+            raise ClientResponseError(url, response)
 
-            if txn["nonce"] < next_nonce and not isExecuted:
-                continue  # NOTE: Skip orphaned transactions
+    def post_signature(
+        self,
+        safe_tx_or_hash: Union[SafeTx, SafeTxID],
+        signer: AddressType,
+        signature: MessageSignature,
+    ):
+        if isinstance(safe_tx_or_hash, get_args(SafeTx)):
+            safe_tx = safe_tx_or_hash
+            safe_tx_hash = hash_eip712_message(safe_tx).hex()
+        elif isinstance(safe_tx_or_hash, get_args(SafeTxID)):
```

Comment:
is `get_args` supposed to be a type here? seems like a typo (currently is failing)